deno_dist/context.ts

```diff
@@ -48,9 +48,40 @@ interface JSONTRespond {
     object: T extends JSONValue ? T : JSONValue,
     status?: StatusCode,
     headers?: HeaderRecord
-  ): TypedResponse<T extends JSONValue ? (JSONValue extends T ? never : T) : never>
+  ): TypedResponse<
+    InterfaceToType<T> extends JSONValue
+      ? JSONValue extends InterfaceToType<T>
+        ? never
+        : T
+      : never
+  >
+  <T>(
+    object: InterfaceToType<T> extends JSONValue ? T : JSONValue,
+    status?: StatusCode,
+    headers?: HeaderRecord
+  ): TypedResponse<
+    InterfaceToType<T> extends JSONValue
+      ? JSONValue extends InterfaceToType<T>
+        ? never
+        : T
+      : never
+  >
   <T>(object: T extends JSONValue ? T : JSONValue, init?: ResponseInit): TypedResponse<
-    T extends JSONValue ? (JSONValue extends T ? never : T) : never
+    InterfaceToType<T> extends JSONValue
+      ? JSONValue extends InterfaceToType<T>
+        ? never
+        : T
+      : never
+  >
+  <T>(
+    object: InterfaceToType<T> extends JSONValue ? T : JSONValue,
+    init?: ResponseInit
+  ): TypedResponse<
+    InterfaceToType<T> extends JSONValue
+      ? JSONValue extends InterfaceToType<T>
+        ? never
+        : T
+      : never
   >
```

Comment:
I had to add overloads for `JSONTRespond` to avoid breaking existing test cases.
(I had to define both `object: T extends JSONValue ? T : JSONValue` and `object: InterfaceToType<T> extends JSONValue ? T : JSONValue` versions.)